### PR TITLE
Fix hooks. Issue #73

### PR DESCRIPTION
**Pull request by matiasdelellis**
Hi Branko,
Finally I could fix it. :sleepy: 

I will comment on the same commits .. :wink:

### CODE REVIEW

File: lib/Watcher.php [hunk 1]
```diff
@@ -101,10 +101,11 @@ public function __construct(IConfig       $config,
 	public function postWrite(Node $node) {
 		$model = intval($this->config->getAppValue('facerecognition', 'model', AddDefaultFaceModel::DEFAULT_FACE_MODEL_ID));
 
+		// FIXME: Can we know if it is a local or shared file?. See issue #73 on Github.
 		// todo: should we also care about this too: instanceOfStorage(ISharedStorage::class);
-		if ($node->getStorage()->instanceOfStorage(IHomeStorage::class) === false) {
+		/*if ($node->getStorage()->instanceOfStorage(IHomeStorage::class) === false) {
```
[the remark this answers is not in the document]
Comment (stalker314314) [reply lib/Watcher.php]: OMG, let me reedit previous comment, it is compltely wrong! Done, take a look now

File: lib/Watcher.php [hunk 2]
```diff
@@ -116,6 +117,14 @@ public function postWrite(Node $node) {
 			return;
 		}
 
+		// NOTE: The hooks also report the creation of thumbnails, and other files.
+		// The way to differentiate between them, is that the user files starting with the userId, and the thumbains as appdata_ocs######
+		$absPath = ltrim($node->getPath(), '/');
+		$owner = explode('/', $absPath)[0];
+		if (!$this->userManager->userExists($owner)) {
```
Comment (stalker314314) [inline lib/Watcher.php]: this check (that user exists) is OK for local/home files. Not sure how will this look like with shared/external files?

File: lib/Watcher.php [hunk 3]
```diff
@@ -116,6 +117,14 @@ public function postWrite(Node $node) {
 			return;
 		}
 
+		// NOTE: The hooks also report the creation of thumbnails, and other files.
+		// The way to differentiate between them, is that the user files starting with the userId, and the thumbains as appdata_ocs######
+		$absPath = ltrim($node->getPath(), '/');
+		$owner = explode('/', $absPath)[0];
```
Comment (stalker314314) [inline lib/Watcher.php]: concept of "owner" exist for local files only. OK, maybe for shared? But for external files, it does not make sense, I think? Dunno, never studied that

File: appinfo/info.xml [hunk 4]
```diff
@@ -15,6 +15,9 @@
 	<website>https://github.com/matiasdelellis/facerecognition</website>
 	<bugs>https://github.com/matiasdelellis/facerecognition/issues</bugs>
 	<repository type="git">https://github.com/matiasdelellis/facerecognition.git</repository>
+	<types>
```
Comment (stalker314314) [inline appinfo/info.xml]: so, this was main issue?:)

**Comment by matiasdelellis** [comment]
D'Oh!. I never summit my review with the comments.. Please see again. That most of your doubts are answered.

File: appinfo/info.xml (hunk 4 above)
Comment (matiasdelellis) [reply appinfo/info.xml]: Yes.. Please, look my comments that I just summit..

File: lib/Watcher.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (matiasdelellis) [reply lib/Watcher.php]: But if having "Deleted files" activated, the storage is reported as: "OCA\Files_Trashbin\Storage" no matter what storage is behind it, we have no way of knowing if it is Shared/federated/external storage.. :disappointed:
Try searching for storage->getParent (), but you can not access it.. Therefore we have to look for alternatives ..

File: lib/Watcher.php (hunk 3 above)
Comment (matiasdelellis) [reply lib/Watcher.php]: I do not know, but look at my comments that until recently were pending ....

File: appinfo/info.xml [hunk 5]
```diff
@@ -15,6 +15,9 @@
...
+		<filesystem/>
+	</types>
```
[the remark this answers is not in the document]
Comment (stalker314314) [reply appinfo/info.xml]: no, this is really awesome that you found this, kudos man!

P.S. another reason to say that minimum version is 14, IMHO

File: lib/Watcher.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (stalker314314) [reply lib/Watcher.php]: So, let me see if I understand you: we get Trashbin **only**, **no matter** if we have `<types>` in `info.xml` or not?

Or - we get Trashbin storage type here, **but** we also do have other storage types, if `<types>` is in info.xml? I need to test on my machine when I find time

File: lib/Watcher.php (hunk 3 above)
Comment (stalker314314) [reply lib/Watcher.php]: ok, got it - so, you basically detect storage type, based on `getPath()` ?:) This miiiight work, if we do not have other options:)

File: appinfo/info.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (matiasdelellis) [reply appinfo/info.xml]: Yes. I think the same .. It already left NC 15 .. :sweat_smile: 
But I do not update to make sure it works in 14.. :wink:

File: lib/Watcher.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (matiasdelellis) [reply lib/Watcher.php]: Without <types>, it does not report the user's files. Strangely, the creation of the thumbnails is reported. :confused:

Regardless of the <typos>, every time the hooks report, the storage says Trashbin.

Please, when you can corroborate it.. :wink:

**Comment by matiasdelellis** [comment]
I merge it ..
It can improve, but let's discuss and test point by point later. :wink:

**Pull request merged**